### DISCUSSION TITLE
Add a way to run a turbopack loader differently on client and server

### DISCUSSION THREAD

**Discussion by remorses** [vercel/next.js#63150] [Ideas]
### Goals

1. Previously you could run a source transformation differently if the file was loaded on the server or the client, now this is not possible anymore with the `experimental.turbo.loaders` option

### Background

I want to turn some Next.js plugins into turbopack loaders (https://github.com/remorses/elacca and https://github.com/Janpot/next-rpc) but it's not possible if you can't run the loader differently on server and client

### Proposal

Run the loader 2 times for server and client, add a field `isServer` to the loader context

```ts
import type { LoaderThis } from './types';

/**
 * Test loader for turbopack
 */
export default function (this: LoaderThis<unknown>, source: string, map: any): void {
  this.async();

  const isServer = this.isServer

  this.callback(null, source, map);
}
```

**Comment by sokra** [comment]
You can use conditions for loader rules now:

```
turbo: {
  rules: {
    "*.blah": {
      browser: {
        loaders: [...]
      },
      default: {
        loaders: [...]
      }
    }
  }
}
```

You can just a bunch of different conditions (like `browser`, `development`, `production`)
